Upgrade `go-github` to newer version; code and test cleanups

DiSiqueira opened the pull request:
- Modify `gofmt_test.sh` to use the simplify argument on gofmt
- Executes gofmt with the simplify argument in all files
- [Remove unnecessary import rename](https://github.com/golang/go/wiki/CodeReviewComments#imports)
- Reformat all imports using `goimports -local github.com/google/`
- Uses context.Background() instead of oauth2.NoContext (Deprecated)
- Update the code to work with the new version of github package.
- Remove unused constant
- Underscore unused parameter
- Fix a simple typo
- [Error strings should not be capitalized](https://github.com/golang/go/wiki/CodeReviewComments#error-strings)

mbrukman commented:
@DiSiqueira — thanks for the cleanups! I'm sorry, I didn't even get to add the Travis CI integration, so this PR isn't even tested yet — I filed issue #2 for myself to do this, but you beat to it — that was supposed to be task #1!

Let me set up Travis first, and then we'll get this PR tested and merged. Thanks for your patience.

DiSiqueira commented:
No problems @mbrukman! If you need anything else, just let me know!

mbrukman commented:
@DiSiqueira — thanks for your patience, and apologies for the (very long) delay! I've been thinking for a while which specific vendoring approach to take with Go, given that there hasn't been a single winner, nothing built-in (besides the Go 1.5 `vendor` experiment, which didn't seem like it was going to go mainstream) and many third-party attempts.

I've settled on Go 1.11's `go mod` which is now used in the repo, and I've added Travis CI testing which verifies that this works. Could you please rebase your PR on the latest changes in the `master` branch, update your Go installation to 1.11.x if you're not running on it already, and update `go.sum` to use a more recent version of the `go-github` package which is what you've fixed the context issues for?

Thanks again for all your cleanups! I really appreciate your fixes and cleanups and look forward to merging this PR.

mbrukman commented:
@DiSiqueira — one more thing: if you run tests with `make test` and not simply `go test`, note that the tests depend on a specific version of `gomock` and `mockgen` tools as they compare golden output in the repo vs. what's generated by the tool; see the README for how to install the versions that will be compatible with the current code base, or just re-run the tests on Travis CI after you force-push the branch, which will use the correct versions.

DiSiqueira commented:
Thanks for the feedback @mbrukman !! I rebased my branch and updated the `go-github` package to the latest version. All the tests seem to be working, so when you have a minute, please review it again.

mbrukman commented on go.sum:
```diff
@@ -5,23 +5,32 @@ github.com/golang/mock v1.2.0 h1:28o5sBqPkBsMGnC6b4MvE2TzSr5/AT4c/1fLqVGIwlk=
 github.com/golang/protobuf v1.2.0/go.mod h1:6lQm79b+lXiMfvg/cZm0SGofjICqVBUtrP5yJMmIC1U=
-github.com/google/go-github v3.0.0+incompatible h1:lsf+ZqlC8Jm53WVfjFavvCCIamHR13YTWhhkB9j549Y=
-github.com/google/go-github v3.0.0+incompatible/go.mod h1:zLgOLi98H3fifZn+44m+umXrS52loVEgC2AApnigrVQ=
+github.com/google/go-github v17.0.0+incompatible h1:N0LgJ1j65A7kfXrZnUDaYCs/Sf4rEjNlfyDHW9dolSY=
```
Please run `go mod tidy` which should delete the v17 lines and anything else that's no longer needed.

DiSiqueira replied:
`go mod tidy` does not remove this line, even if I delete the `go.sum` file and re-generate it, this line is included again, and if I remove it manually any `go mod` related command like `go get` will also include it again.

mbrukman replied:
Interesting, I guess it's a transitive dependency for one of our dependencies or something else. Thanks for looking into it!

mbrukman commented on ghutil/mock_ghutil.go:
```diff
@@ -19,8 +19,9 @@
 package ghutil
 
 import (
+	context "context"
```
I think you cleaned up other imports; should this just be `context`?

DiSiqueira replied:
MockGen generates this file (I re-generated it using `make mock`), I can manually clean up the imports here, but it will cause the `make test` on this folder to fail. 
Same for the others.

mbrukman replied:
My apologies; you're absolutely right, I missed that this was a generated file.

mbrukman commented on ghutil/mock_ghutil.go:
```diff
@@ -19,8 +19,9 @@
 
 import (
+	context "context"
 	gomock "github.com/golang/mock/gomock"
```
Should we drop the initial `gomock` as it's superfluous?

mbrukman replied:
Generated file; this is not feasible.

mbrukman commented on ghutil/mock_ghutil.go:
```diff
@@ -19,8 +19,9 @@
 	gomock "github.com/golang/mock/gomock"
-	github "github.com/google/go-github/github"
+	github "github.com/google/go-github/v21/github"
 	reflect "reflect"
```
Should this just be `reflect`?

mbrukman replied:
Generated file; this is not feasible.
